rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -73,20 +78,29 @@
     @NonNull
     private final RestControllerHelper restControllerHelper;
 
+    @NonNull
+    private final ResourceListController resourceListController = new ResourceListController();
+
     @RequestMapping(value = COMPONENTS_URL, method = RequestMethod.GET)
-    public ResponseEntity<Resources<Resource<Component>>> getComponents(@RequestParam(value = "name", required = false) String name,
+    public ResponseEntity<Resources<Resource<Component>>> getComponents(Pageable pageable,
+                                                                        @RequestParam(value = "name", required = false) String name,
                                                                         @RequestParam(value = "type", required = false) String componentType,
                                                                         @RequestParam(value = "fields", required = false) List<String> fields,
-                                                                        OAuth2Authentication oAuth2Authentication) throws TException {
+                                                                        OAuth2Authentication oAuth2Authentication) throws TException, URISyntaxException, WrongPaginationParameterException, ResourceClassNotFoundException {
 
         User sw360User = restControllerHelper.getSw360UserFromAuthentication(oAuth2Authentication);
-        List<Component> sw360Components = new ArrayList<>();
+
+        List<Component> allComponents = new ArrayList<Component>();
         if (name != null && !name.isEmpty()) {
-            sw360Components.addAll(componentService.searchComponentByName(name));
+            allComponents.addAll(componentService.searchComponentByName(name));
         } else {
-            sw360Components.addAll(componentService.getComponentsForUser(sw360User));
+            allComponents.addAll(componentService.getComponentsForUser(sw360User));
         }
 
+        PaginationOptions paginationOptions = restControllerHelper.paginationOptionsFromPageable(pageable, new Component().getType());
```

Comment:
If `pageable` is null you will call the complete method sequence for pagination.
You could verify here if you need to page / filter the result.

Reply:
I think `pageable` is always `!=NULL` at this point, because there's a defined default value for `page_entries` in `Sw360ResourceServer`. Even if there are no paging parameters in the request, the pageable object will contain `page=0` and `page_entries=10`